light/errors.go

```diff
@@ -40,11 +201,229 @@ func (e ErrInvalidHeader) Error() string {
+	return e.Err
+}
+
+type ErrHeaderValidateBasic struct {
+	Err error
+}
+
+func (e ErrHeaderValidateBasic) Error() string {
+	return fmt.Sprintf("untrustedHeader.ValidateBasic failed: %v", e.Err)
+}
+
+func (e ErrHeaderValidateBasic) Unwrap() error {
+	return e.Err
+}
+
+type ErrInvalidTrustOptions struct {
+	Err error
+}
+
+func (e ErrInvalidTrustOptions) Error() string {
+	return fmt.Sprintf("invalid TrustOptions: %v", e.Err)
+}
+
+func (e ErrInvalidTrustOptions) Unwrap() error {
+	return e.Err
+}
+
+type ErrGetTrustedBlock struct {
+	Err error
+}
+
+func (e ErrGetTrustedBlock) Error() string {
+	return fmt.Sprintf("can't get last trusted light block: %v", e.Err)
+}
+
+func (e ErrGetTrustedBlock) Unwrap() error {
+	return e.Err
+}
+
+type ErrGetTrustedBlockHeight struct {
+	Err error
+}
+
+func (e ErrGetTrustedBlockHeight) Error() string {
+	return fmt.Sprintf("can't get last trusted light block height: %v", e.Err)
+}
+
+func (e ErrGetTrustedBlockHeight) Unwrap() error {
+	return e.Err
+}
+
+type ErrCleanup struct {
+	Err error
+}
+
+func (e ErrCleanup) Error() string {
+	return fmt.Sprintf("failed to cleanup: %v", e.Err)
+}
+
+func (e ErrCleanup) Unwrap() error {
+	return e.Err
+}
+
+type ErrGetBlock struct {
+	Err error
+}
+
+func (e ErrGetBlock) Error() string {
+	return fmt.Sprintf("failed to retrieve light block from primary to verify against: %v", e.Err)
+}
+
+func (e ErrGetBlock) Unwrap() error {
+	return e.Err
+}
+
+type ErrGetFirstBlock struct {
+	Err error
+}
+
+func (e ErrGetFirstBlock) Error() string {
+	return fmt.Sprintf("can't get first light block: %v", e.Err)
+}
+
+func (e ErrGetFirstBlock) Unwrap() error {
+	return e.Err
+}
+
+type ErrGetFirstBlockHeight struct {
+	Err error
+}
+
+func (e ErrGetFirstBlockHeight) Error() string {
+	return fmt.Sprintf("can't get first light block height: %v", e.Err)
+}
+
+func (e ErrGetFirstBlockHeight) Unwrap() error {
+	return e.Err
+}
+
+type ErrInvalidCommit struct {
+	Err error
+}
+
+func (e ErrInvalidCommit) Error() string {
+	return fmt.Sprintf("invalid commit: %v", e.Err)
+}
+
+func (e ErrInvalidCommit) Unwrap() error {
+	return e.Err
+}
+
+type ErrGetLastTrustedHeight struct {
+	Err error
+}
+
+func (e ErrGetLastTrustedHeight) Error() string {
+	return fmt.Sprintf("can't get last trusted height: %v", e.Err)
+}
+
+func (e ErrGetLastTrustedHeight) Unwrap() error {
+	return e.Err
+}
+
+type ErrPrune struct {
+	Err error
+}
+
+func (e ErrPrune) Error() string {
+	return fmt.Sprintf("prune: %v", e.Err)
+}
+
+func (e ErrPrune) Unwrap() error {
+	return e.Err
+}
+
+type ErrSaveTrustedHeader struct {
+	Err error
+}
+
+func (e ErrSaveTrustedHeader) Error() string {
+	return fmt.Sprintf("failed to save trusted header: %v", e.Err)
+}
+
+func (e ErrSaveTrustedHeader) Unwrap() error {
+	return e.Err
+}
+
+type ErrCleanupAfter struct {
+	Height int64
+	Err    error
+}
+
+func (e ErrCleanupAfter) Error() string {
+	return fmt.Sprintf("cleanupAfter(%d): %v", e.Height, e.Err)
```

Comment:
This does not look too human-readable.
```suggestion
	return fmt.Sprintf("cleanup after height %d failed: %v", e.Height, e.Err)
```